pandas/io/parsers.py

```diff
@@ -272,6 +275,28 @@
 """ % (_parser_params % (_fwf_widths, ''))
 
 
+def _validate_nrows(nrows):
+    """
+    Checks whether the 'nrows' parameter for parsing is either
+    an integer OR float that can SAFELY be cast to an integer
+    without losing accuracy. Raises a ValueError if that is
+    not the case.
+    """
+    msg = "'nrows' must be an integer or a safely castable float"
+
+    if nrows is not None:
+        if not com.is_integer(nrows):
+            if com.is_float(nrows):
+                if int(nrows) != nrows:
+                    raise ValueError(msg)
+                else:
```

Comment:
The super-nesting of conditionals can probably be changed.  We just need to make sure that `nrows` is an `int` (perhaps in `float` form).